src/app/modules/form/public-form/public-form.controller.ts

```diff
@@ -161,3 +171,141 @@ export const handleRedirect: RequestHandler<
     redirectPath,
   })
 }
+
+/**
+ * Handler for GET /:formId/publicform endpoint
+ * @returns 200 if the form exists
+ * @returns 404 if form with formId does not exist or is private
+ * @returns 410 if form has been archived
+ * @returns 500 if database error occurs or if the type of error is unknown
+ */
+export const handleGetPublicForm: RequestHandler<{ formId: string }> = async (
+  req,
+  res,
+) => {
+  const { formId } = req.params
+
+  const formResult = await getFormIfPublic(formId).andThen((form) =>
+    FormService.checkFormSubmissionLimitAndDeactivateForm(form),
+  )
+
+  // Early return if form is not public or any error occurred.
+  if (formResult.isErr()) {
+    const { error } = formResult
+    logger.error({
+      message: 'Error retrieving public form',
+      meta: {
+        action: 'handleGetPublicForm',
+        ...createReqMeta(req),
+        formId,
+      },
+      error,
+    })
+    const { errorMessage, statusCode } = mapRouteError(error)
+    return res.status(statusCode).json({ message: errorMessage })
+  }
+
+  const form = formResult.value
+  const publicFormView = form.getPublicView()
+  const { authType } = form
+
+  // NOTE: Once there is a valid form retrieved from the database,
+  // the client should always get a 200 response with the form's public view.
+  // Additional errors should be tagged onto the response object like myInfoError.
+  switch (authType) {
+    case AuthType.SP:
+    case AuthType.CP: {
+      // Form is valid, check for SPCP/MyInfo data.
+      return SpcpFactory.getSpcpSession(authType, req.cookies)
+        .map(({ userName }) =>
+          res.json({
+            form: publicFormView,
+            spcpSession: { userName },
+          }),
+        )
+        .mapErr((error) => {
+          // NOTE: Only log if there is no jwt present on the request.
+          // This is because clients can be members of the pubilc and hence, have no jwt.
+          if (!(error instanceof MissingJwtError)) {
+            logger.error({
+              message: 'Error getting public form',
+              meta: {
+                action: 'handleGetPublicForm',
+                ...createReqMeta(req),
+                formId,
+              },
+              error,
+            })
+          }
+          return res.json({
+            form: publicFormView,
+          })
+        })
+    }
+
+    case AuthType.MyInfo: {
+      const requestedAttributes = form.getUniqueMyInfoAttrs()
```

Comment:
this is only for logging purposes, right? I don't think it's necessary, since they will be logged in the service layer, and it's not strictly necessary information anyway